### ISSUE TITLE
[Logs] Resolve stderr logging problem with DD

### ISSUE THREAD

**Issue by robrap** [edx/edx-arch-experiments#591]
DD logging has two parts: 1) ingest, and 2) indexing. Hopefully we don't need to pay be either for log messages that we don't need.

The main priority of this task is to understand our actual logging numbers for the DD contract, while at the same time protecting NR and Splunk from unnecessary log data and additional expense.

**Acceptance Criteria-sh:**

- [ ] Discover the major sources of the log issue.
  - [ ] Is this due to NR+DD, or other?
- [ ] Add additional drop log filters in NR (as-needed)
- [ ] Add additional drop log filters in Splunk (as-needed)
- [ ] Determine a game plan for resolving. 
  - [ ] Does it need to wait until we can disable NR? 
  - [ ] How can DD ignore this log data (if we don't resolve) for the contract ingest numbers?
- [ ] Engage with DD Slack thread to determine if disabling on Edge will be a good experiment, and potentially creating a support ticket.
- [ ] Determine if mongodb prod issue and support ticket are related, or should be spun into its own separate ticket.

**Notes:**

- See [last 3 days of logging gigs in NR](https://onenr.io/0PwJ57zg7Q7), or [last 14 days](https://onenr.io/0ERPqmODDjW) to get high-level view of the problem from NR side. Logging data increased ~3x.
- See related [log drop filters in New Relic](https://one.newrelic.com/logger/log-drop-filters?account=88178&duration=1800000&state=58ed922f-aa4d-d0b4-2433-b4244d5c1c30).
  - This doesn't seem to have made a large dent in the logging bytes in NR.
  - How many more types of messages, and can we pick this off one-by-one or will this take forever?
  - See one [example message over the last 7 days](https://onenr.io/07wk7aDJnwL). This query is slow and probably costly. You can see where config changes when into affect on April 3 and April 4 for discovery prod, which is the biggest source of the messages.
- [question] Are the issues related to DD config issues, DD bugs, NR+DD monitoring interference, or other?
  - Disabling NR in Edge is one experiment, but we'll only have DD logs to verify, and we'd need to have specific messages that we see go away.
- [Slack thread with DD about this issue](https://twou.slack.com/archives/C06QEAJHLC9/p1712611896958199).
  - Includes the 4 messages filtered so far (as of Apr 8).
- [Longer Slack internal thread](https://twou.slack.com/archives/C06DMDFNBJQ/p1712608185230359?thread_ts=1712263187.875359&cid=C06DMDFNBJQ) with more links and charts from NR.
- Is it possible that this is the source of the mongodb disk issue? See https://help.datadoghq.com/hc/en-us/requests/1635160 (using Google login).

**Comment by robrap** [comment]
See [DD Slack thread comment](https://twou.slack.com/archives/C06QEAJHLC9/p1712685857186229?thread_ts=1712611896.958199&channel=C06QEAJHLC9&message_ts=1712685857.186229).
- I had an idea to make New Relic go dark for some Stage app that has these log messages coming up as errors.
  - It might be nice to get Alex R to avoid filtering these messages in DD so we have more history than 15 minute live tal, but we also don't want to be blocked.
- Someone needs to follow-up with DeVaughn, because he's making it seem like ecommerce (where these aren't errors, but are logs), includes these messages where it may be expected? But that is a ton of log data (at least relative to what we used to have in NR).

**Comment by robrap** [comment]
Noting separately that there are related APM tracing issues that need to be resolved, and are indicated by these messages.

**Comment by robrap** [comment]
Additionally, on the [DD support ticket for the mongodb issue](https://help.datadoghq.com/hc/en-us/requests/1635160), DD suggested that we could:
> Can you try setting the Datadog log level to WARN and see if that helps
> This is done on the datadog.yaml file
> set the logs_level: WARN

We **may** want to do this, but first let's do the above stage credentials change to see what happens.

**Comment by timmc-edx** [comment]
I've disabled NR in stage credentials, deployed as of 18:08 UTC: https://github.com/edx/edx-internal/pull/10610

**Comment by jristau1984** [comment]
An experiment was done with credentials, which did not produce the result desired... the logs remain unchanged in DataDog.  We have logged a DataDog request to investigate.  Blocking this until they come back with something:
https://help.datadoghq.com/hc/en-us/requests/1643101

**Comment by robrap** [comment]
@jristau1984 @timmc-edx: Consider temporarily disabling DD APM in prod discovery to remove some/much? of the logging noise going to splunk and new relic.

**Comment by robrap** [comment]
Note: The mongodb support ticket has some requests for additional logs (e.g. debug, etc.): https://help.datadoghq.com/hc/en-us/requests/1635160. Although we can wait on the new support ticket, we might try enabling debug logs in one stage service (if possible) for a few minutes, etc. Anyway, if these tickets aren't linked on the DD-side, we should probably link them.

**Comment by dianakhuang** [comment]
We have found a setting for dd-trace that reduced/removed the logging messages being thrown by dd-trace on our k8s clusters to a reasonable level. We have also updated the log level to WARN for most of our EC2 environments. Doing some testing now on ecommerce to see if this setting does get applied correctly.

**Comment by timmc-edx** [comment]
Details: https://github.com/edx/helm-charts/pull/122 (DD_TRACE_LOG_STREAM_HANDLER=false)

**Comment by robrap** [comment]
@timmc-edx: Can you ticket the potential alternative and any concerns with the stability of the current solution with changelog, etc. Update the support ticket as needed as well. Then close this ticket. Thanks!

**Comment by timmc-edx** [comment]
Created https://github.com/edx/edx-arch-experiments/issues/610 for future investigation.
